docs/modules/configuring/con-kafkaproxy-resource-allocation.adoc

```diff
@@ -0,0 +1,39 @@
+// file included in the following:
+//
+// kroxylicious-operator/assemblies/assembly-operator-operate-resource-allocation.adoc
+
+[id='con-kafkaproxy-tuning-resource-allocation-{context}']
+== Configuring Proxy container CPU and memory resource limits and requests
+
+When you define a `KafkaProxy` specification, a Kubernetes Deployment of Proxy containers is created. Each of these containers runs a single Kroxylicious Proxy process.
+
+By default, these Proxy containers are configured with the following resource requests and limits:
+
+- `resources.requests.cpu`: `500m`
+- `resources.requests.memory`: `512Mi`
+- `resources.limits.cpu`: `500m`
+- `resources.limits.memory`: `512Mi`
+
+You can learn more about Kubernetes resource management https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/[here].
+
+To replace these default resource allocations, modify the `proxyContainer` section within your `KafkaProxy` specification.
```

Comment:
Is it worth clarifying that requests and limits are independent? I.e. users could choose to expands requests but not limits (or is it the other way round) to allow the proxy to cope with bursty traffic?

Reply:
other way around, requests is a minimum requirement for scheduling and limits is the max it can use. 

it's tricky to word but I think we need to somehow convey that if they specify any `resources` it replaces the defaults entirely. For example the user can entirely remove the limits by specifying:

``` yaml
resources:
  requests:
    cpu: 600m
    memory: 700Mi
```

Reply:
Ohh. Sad but fair enough.

Reply:
Chatting with @tombentley we think we should update the operator to perform a merge. So that if the user doesn't specify a value they get the operator defaults and there is no way to say use what ever you like.

Reply:
If going that way maybe we could put defaults on the CRD, then it would be visible there too?